accepted/00000-salt-ssh-with-salt-bundle.md

```diff
@@ -0,0 +1,95 @@
+
+This approach will help us to limit the number of supported versions of Salt codebase and drop `py26-compat-salt`, `py27-compat-salt` (as these packages are shipping `2016.11.10` and `3000.3` respectively) and all Python 2 dependencies maintenance.
+So we don't need to maintain old salt codebases, but focus only on the latest one.
+
+# Detailed design
+[design]: #detailed-design
+
+## Uyuni roster
+
+We can use Uyuni roster module to provide roster data instead of creating flat roster files with Java. PR moving the logic to Uyuni roster on Java side: https://github.com/uyuni-project/uyuni/pull/4457
+
+This change can help to avoid creating temporary roster files (https://docs.saltproject.io/en/latest/topics/ssh/roster.html) and implements caching additionally.
+
+## Pre flight script
+
+Salt can execute a shell script on the salt-ssh client before the "real salt-ssh execution" starts.
+This script downloads the salt-bundle package from the appropriate bootstrap repository and extracts the virtual environment.
+By-default there is no way to pass additional parameters to pre fligt script running on the minion.
+In this case we have to generate individual pre flight script to each of the minion. The main drawback here is maintaining these pre flight script files.
+
+The other solution is to pass additional parameter to the static pre flight script.
+The pre flight script calculates the bootstrap repository URL in a similar manner to the existing boostrap.sh script, based on the operating system and OS architecture of the system it is executed on.
+It requires one parameter from the Uyuni server: host, port (entry point) for different contact methods and pathes (SSH from server/proxy, direct access to the repos or with the tunnel).
+The salt codebase need to be modified to pass additional parameters, therwise we have to generate pre flight script individually for each minion.
+
+## Deploying Salt Bundle to the minion
+
+We don't need to install Salt Bundle package on the minion.
+We need to extract the virtual environment of the bundle from the package.
+Pre flight script is deploying it as described above.
+
+## Using proper salt codebase
+
+On deploying Salt Bundle with pre flight script, without modification salt-thin (`salt-thin` is a tarball with salt codebase created by the server to deploy salt on the salt-ssh minion: https://docs.saltproject.io/en/latest/ref/runners/all/salt.runners.thin.html) deployment, we will get two different codebases of salt deployed on the minion (one codebase is from salt bundle package for the OS+arch and the other one is from salt-master delivered with salt-thin tarball).
+As the sources of these two codebases are different (update channel of the Server and client tools relevant for the system) these codebases are not in sync.
+We can prevent deploying salt-thin in case if Salt Bundle has been deployed already and use it as preferred salt codebase to handle salt-ssh session with on the minion side.
```

Comment:
As this is an RFC, you should be assertive in your proposal - the intent here is to describe what the problem is, how you would solve it and why that way is better than others rather than just proposing alternatives.

"We can" is not an assertion - your sentences should sound like "X will happen", "X will not happen" and so on (with justifications).

In this case: assuming the RFC is completely and correctly implemented, as a user, will I see salt-thin being used or not? If so, at what conditions?